processing/src/main/java/io/druid/query/groupby/epinephelinae/GroupByMergingQueryRunnerV2.java

```diff
@@ -148,6 +134,11 @@ public GroupByMergingQueryRunnerV2(
         String.format("druid-groupBy-%s_%s", UUID.randomUUID(), query.getId())
     );
 
+    final LimitedTemporaryStorage temporaryStorage = new LimitedTemporaryStorage(
```

Comment:
I feel that it's more appropriate for the temporary storage to be instantiated together with the temporaryStorageDirectory. LimitedTemporaryStorage instantiation currently does not do any resource allocation similar to java File.